runtime/pkg/globutil/globutil_test.go

```diff
@@ -0,0 +1,49 @@
+package globutil
+
+import "testing"
+
+func TestParseURL(t *testing.T) {
+	tests := []struct {
+		name    string
+		args    string
+		want    string
+		want1   string
+		want2   string
+		wantErr bool
+	}{
+		{
+			name:    "valid s3",
+			args:    "s3://bucket.rill-developer.io/**/path_?/{0,1,2}000016[1-2]/*.parquet",
+			want:    "s3",
+			want1:   "bucket.rill-developer.io",
+			want2:   "**/path_?/{0,1,2}000016[1-2]/*.parquet",
+			wantErr: false,
+		},
+		{
+			name:    "invalid s3",
+			args:    "s3:/bucket.rill-developer.io/**/path_?/{0,1,2}000016[1-2]/*.parquet",
+			want:    "",
+			want1:   "",
+			want2:   "",
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, got1, got2, err := ParseURL(tt.args)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ParseURL() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("ParseURL() got = %v, want %v", got, tt.want)
+			}
+			if got1 != tt.want1 {
+				t.Errorf("ParseURL() got1 = %v, want %v", got1, tt.want1)
+			}
+			if got2 != tt.want2 {
+				t.Errorf("ParseURL() got2 = %v, want %v", got2, tt.want2)
+			}
+		})
```

Comment:
1. Use `require.NotEqual` and similar
2. Maybe consider not doing a matrix test for such few cases – has a readability loss

Reply:
I am not aware of the term matrix test so not sure what it means here. Changed the struct field names for better readability. Anyways both the actual code and test code is small and easy to read/understand.

Reply:
Not sure it's called a matrix test, just what popped into my mind for the pattern of looping over a list of test case params. Is your IDE generating these for you? 

When there are few cases, I sometimes find them harder to read than something like [this](https://github.com/rilldata/rill-developer/blob/main/runtime/drivers/duckdb/config_test.go). But they're very common and this is just a personal preference. So no need to change.